src/main/java/com/cloudbees/jenkins/plugins/bitbucket/impl/extension/BitbucketEnvVarExtension.java

```diff
@@ -0,0 +1,39 @@
+package com.cloudbees.jenkins.plugins.bitbucket.impl.extension;
+
+import com.cloudbees.jenkins.plugins.bitbucket.impl.util.URLUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import java.util.Map;
+
+public class BitbucketEnvVarExtension extends GitSCMExtension {
+
+    private final String owner;
+    private final String repository;
+    private final String projectKey;
+    private final String serverURL;
+
+    public BitbucketEnvVarExtension(@Nullable String owner, @NonNull String repository, @Nullable String projectKey, @NonNull String serverURL) {
```

Comment:
You are missing getters, and `@DataBoundConstructor`, and a `Descriptor`. Does this actually work? It does not look like it would.

Reply:
It works, also other GitSCMExtension in this plugin works without @DataBoundConstructor
Anyway In master I have added this annotation and descriptor.

Reply:
I think it works because XStream is able to build the object using reflection without use constructor (unsafe class) and inject fields

Reply:
FTR b777c1be6f09d47fde507d0603e1ca5817dbc740 & 7c1c340bcb298089d7cf4a40515c7dca88f76aa5

Reply:
As example the GitClientAuthenticatorExtension won't work with `@DataBoundConstructor`. It fails with
`Could not instantiate arguments for com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl. Secrets are involved, so details are available on more verbose logging levels.`